DataFormats/PatCandidates/interface/Muon.h

```diff
@@ -32,400 +33,413 @@
+    /// if force == false, do not embed this track if it's embedded already (e.g. ig it's a tracker track, and that's already embedded)
+    void embedMuonBestTrack(bool force = false);
+    /// set reference to Track selected to be the best measurement of the muon parameters (reimplemented from reco::Muon)
+    /// if force == false, do not embed this track if it's embedded already (e.g. ig it's a tracker track, and that's already embedded)
+    void embedTunePMuonBestTrack(bool force = false);
+    /// set reference to Track reconstructed in the tracker only (reimplemented from reco::Muon)
+    void embedTrack();
+    /// set reference to Track reconstructed in the muon detector only (reimplemented from reco::Muon)
+    void embedStandAloneMuon();
+    /// set reference to Track reconstructed in both tracked and muon detector (reimplemented from reco::Muon)
+    void embedCombinedMuon();
+
+    // ---- methods for MuonMETCorrectionData ----
+    /// muon MET corrections for caloMET; returns the muon correction struct if embedded during pat tuple production or an empty element
+    reco::MuonMETCorrectionData caloMETMuonCorrs() const {
+      return (embeddedCaloMETMuonCorrs_ ? caloMETMuonCorrs_.front() : reco::MuonMETCorrectionData());
+    };
+    void embedCaloMETMuonCorrs(const reco::MuonMETCorrectionData& t);
+    /// muon MET corrections for tcMET; returns the muon correction struct if embedded during pat tuple production or an empty element
+    reco::MuonMETCorrectionData tcMETMuonCorrs() const {
+      return (embeddedTCMETMuonCorrs_ ? tcMETMuonCorrs_.front() : reco::MuonMETCorrectionData());
+    };
+    void embedTcMETMuonCorrs(const reco::MuonMETCorrectionData& t);
+
+    // ---- methods for TeV refit tracks ----
+
+    /// reference to Track reconstructed using hits in the tracker + "good" muon hits (reimplemented from reco::Muon)
+    reco::TrackRef pickyTrack() const override;
+    /// reference to Track reconstructed using hits in the tracker + info from the first muon station that has hits (reimplemented from reco::Muon)
+    reco::TrackRef tpfmsTrack() const override;
+    /// reference to Track reconstructed using DYT algorithm
+    reco::TrackRef dytTrack() const override;
+    /// Deprecated accessors to call the corresponding above two functions; no dytMuon since this naming is deprecated.
+    reco::TrackRef pickyMuon() const { return pickyTrack(); }  // JMTBAD gcc deprecated attribute?
+    reco::TrackRef tpfmsMuon() const { return tpfmsTrack(); }  // JMTBAD gcc deprecated attribute?
+    /// embed reference to the above picky Track
+    void embedPickyMuon();
+    /// embed reference to the above tpfms Track
+    void embedTpfmsMuon();
+    /// embed reference to the above dyt Track
+    void embedDytMuon();
+
+    // add extra timing information
+    void readExtraTimerInfo(const reco::MuonTimeExtra& t);
```

Comment:
```readTimeExtra``` may be more clear; "timer" is confusing with this word usually used for execution or wall clock time counting.

this method is redundant with setInverseBeta and setInverseBetaErr.
perhaps it's more appropriate to remove setInverseBeta and setInverseBetaErr, in a case there is some need to expand beyond just beta and betaErr.